hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -132,12 +133,10 @@ class ExpressionPayload(@transient record: GenericRecord,
           .serialize(resultingRow)
           .asInstanceOf[GenericRecord]
 
-        if (targetRecord.isEmpty || needUpdatingPersistedRecord(targetRecord.get, resultingAvroRecord, properties)) {
-          resultRecordOpt = HOption.of(resultingAvroRecord)
+        resultRecordOpt = if (targetRecord.isEmpty) {
```

Comment:
the method is `processMatchedRecord`, why we need judge targetRecord is empty?

Reply:
if you check the caller of the method, for MOR table regardless if it is matched record or not we all end up here. I didn't have further context on why the code is written in that way but focusing on fixing the correctness issue of MIT.